atomics/include/desul/atomics/Lock_Array_SYCL.hpp

```diff
@@ -62,9 +62,21 @@ void finalize_lock_arrays_sycl(sycl::queue q);
  * declaration here must be extern). This one instance will be initialized
  * by initialize_host_sycl_lock_arrays and need not be modified afterwards.
  */
-SYCL_EXTERNAL extern sycl_device_global<int32_t*> SYCL_SPACE_ATOMIC_LOCKS_DEVICE;
+#ifdef DESUL_ATOMICS_ENABLE_SYCL_SEPARABLE_COMPILATION
+SYCL_EXTERNAL extern
+#else
+static
```

Comment:
Otherwise, we would have duplicated symbols violating ODR. IIRC, `__device__ __contant__` global variables are implicitly `static` for `CUDA` and `HIP`, also see https://github.com/desul/desul/blob/0a0dfd3a4b53030913f970f0a28714a0415e147a/atomics/include/desul/atomics/Lock_Array_CUDA.hpp#L63-L66 (which means I have to update the comment here as well).

Reply:
2c192da